ludwig/config_sampling/explore_schema.py

```diff
@@ -0,0 +1,304 @@
+
+    Args:
+        explored: deque containing all the config options.
+        config: base Ludwig config to merge the explored configs with.
+    """
+    dataset = build_synthetic_dataset_df(NUM_SYNTHETIC_EXAMPLES, config)
+    ret = []
+    for config_options, _ in explored:
+        for default_config in generate_possible_configs(config_options=config_options):
+            merged_config = merge_dict(copy.deepcopy(config), default_config)
+            try:
+                ModelConfig.from_dict(merged_config)
+                ret.append((merged_config, dataset))
+            except Exception:
+                pass
+    return ret
+
+
+def combine_configs_for_comparator_combiner(
+    explored: Deque[Tuple], config: ModelConfigDict
+) -> List[Tuple[ModelConfigDict, pd.DataFrame]]:
+    """Merge base config with explored sections.
+
+    Completes the entity_1 and entity_2 paramters of the comparator combiner.
+
+    Args:
+        explored: deque containing all the config options.
+        config: base Ludwig config to merge the explored configs with.
+    """
+    dataset = build_synthetic_dataset_df(NUM_SYNTHETIC_EXAMPLES, config)
+    ret = []
+    for item in explored:
+        for default_config in generate_possible_configs(config_options=item[0]):
+            merged_config = merge_dict(copy.deepcopy(config), default_config)
+
+            # create two random lists for entity1 and entity2
+            entity_names = [feature["name"] for feature in config["input_features"]]
+            random.shuffle(entity_names)
+            entity_1_size = random.randint(1, len(entity_names) - 1)
+            merged_config["combiner"]["entity_1"] = entity_names[:entity_1_size]
+            merged_config["combiner"]["entity_2"] = entity_names[entity_1_size:]
+            try:
+                ModelConfig.from_dict(merged_config)
+                ret.append((merged_config, dataset))
+            except Exception:
+                pass
+    return ret
+
+
+def combine_configs_for_sequence_combiner(
+    explored: Deque[Tuple], config: ModelConfigDict
+) -> List[Tuple[ModelConfigDict, pd.DataFrame]]:
+    """Merge base config with explored sections.
+
+    Uses the right reduce_output strategy for the sequence and sequence_concat combiners.
+
+    Args:
+        explored: deque containing all the config options.
+        config: base Ludwig config to merge the explored configs with.
+    """
+    dataset = build_synthetic_dataset_df(NUM_SYNTHETIC_EXAMPLES, config)
+    ret = []
+    for item in explored:
+        for default_config in generate_possible_configs(config_options=item[0]):
+            merged_config = merge_dict(copy.deepcopy(config), default_config)
+            for i in range(len(merged_config["input_features"])):
+                if merged_config["input_features"][i]["type"] in {SEQUENCE, TEXT, TIMESERIES}:
+                    merged_config["input_features"][0]["encoder"] = {"type": "embed", "reduce_output": None}
```

Comment:
This should be set properly now in the ModelConfig's `__post_init__`. Do we still need this combiner-specific generator?

Reply:
I believe we do. This reduces the number of configs that don't pass validation.